### PR TITLE
crush/CrushWrapper.cc: Fix sign compare compiler warning

### PR DESCRIPTION
**Pull request by pdvian**
introduced by 712a39e5c9d

Signed-off-by: Prashant D <pdhange@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

**Comment by pdvian** [comment]
@hjwsm1989 Can you review this PR ?

File: src/crush/CrushWrapper.cc [hunk 1]
```diff
@@ -982,7 +982,7 @@ int CrushWrapper::verify_upmap(CephContext *cct,
           return -EINVAL;
         }
         // validate the osd's in subtree
-        for (int c = 0; cursor < up.size() && c < numrep; ++cursor, ++c) {
+        for (int c = 0; static_cast<uint64_t>(cursor) < up.size() && c < numrep; ++cursor, ++c) {
```
Comment (tchaikov) [inline src/crush/CrushWrapper.cc]: just 2 nits:

- `vector::size()` returns `std::size_t`, and on 32bit platforms, `size_t` is always 32bits. so, probably we can just change the type of `cursor` to `size_t` or `std::size_t` ?
- please wrap at 80 chars. but if the first suggestion makes sense to you, this one is not necessary anymore.

Comment (pdvian) [reply src/crush/CrushWrapper.cc]: Thanks. size_t makes sense here. Updated PR.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2019-11-01_14:35:07-rados-wip-kefu-testing-2019-11-01-1817-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2019-11-03_12:09:54-rados-wip-kefu-testing-2019-11-01-1817-distro-basic-mira/

**Pull request merged**